chipsec/hal/common/ucode.py

```diff
@@ -64,64 +64,60 @@ def __str__(self) -> str:
+            self.logger.log_hal(f'[ucode] CPU{cpu_thread_id:d}: last Microcode update failed (current microcode id = 0x{ucode_update_id:08X})')
         else:
-            logger().log_hal(f'[ucode] CPU{cpu_thread_id:d}: Microcode update ID = 0x{ucode_update_id:08X}')
+            self.logger.log_hal(f'[ucode] CPU{cpu_thread_id:d}: Microcode update ID = 0x{ucode_update_id:08X}')
 
         return ucode_update_id
 
     def update_ucode_all_cpus(self, ucode_file: str) -> bool:
         if not (os.path.exists(ucode_file) and os.path.isfile(ucode_file)):
-            logger().log_error(f"Ucode file not found: '{ucode_file:.256}'")
+            self.logger.log_error(f"Ucode file not found: '{ucode_file:.256}'")
             return False
-        ucode_buf = read_ucode_file(ucode_file)
+        ucode_buf = self.read_ucode_file(ucode_file)
         if (ucode_buf is not None) and (len(ucode_buf) > 0):
-            for tid in range(self.get_cpu_thread_count()):
+            for tid in range(self.cs.hals.Msr.get_cpu_thread_count()):
                 self.load_ucode_update(tid, ucode_buf)
         return True
 
     def update_ucode(self, cpu_thread_id: int, ucode_file: str) -> int:
         if not (os.path.exists(ucode_file) and os.path.isfile(ucode_file)):
-            logger().log_error(f"Ucode file not found: '{ucode_file:.256}'")
+            self.logger.log_error(f"Ucode file not found: '{ucode_file:.256}'")
             return False
-        _ucode_buf = read_ucode_file(ucode_file)
+        _ucode_buf = self.read_ucode_file(ucode_file)
         return self.load_ucode_update(cpu_thread_id, _ucode_buf)
 
     def load_ucode_update(self, cpu_thread_id: int, ucode_buf: AnyStr) -> int:
-        logger().log_hal(f'[ucode] Loading microcode update on CPU{cpu_thread_id:d}')
+        self.logger.log_hal(f'[ucode] Loading microcode update on CPU{cpu_thread_id:d}')
         self.helper.load_ucode_update(cpu_thread_id, ucode_buf)
         return self.ucode_update_id(cpu_thread_id)
+
+    def dump_ucode_update_header(self, pdb_ucode_buffer: bytes) -> UcodeUpdateHeader:
+        if len(pdb_ucode_buffer) < UCODE_HEADER_SIZE:
+            self.logger.log('Buffer is too small')
+            return None
+        ucode_header = UcodeUpdateHeader(*struct.unpack_from('12I', pdb_ucode_buffer))
+        self.logger.log_hal(str(ucode_header))
+        return ucode_header
+
+    def read_ucode_file(self, ucode_filename: str) -> bytes:
+        ucode_buf = read_file(ucode_filename)
+        if (ucode_filename.endswith('.pdb')):
+            self.logger.log_hal(f"[ucode] PDB file '{ucode_filename:256}' has ucode update header (size = 0x{UCODE_HEADER_SIZE:X})")
+            self.dump_ucode_update_header(ucode_buf)
+            return ucode_buf[UCODE_HEADER_SIZE:]
+        else:
+            return ucode_buf
+
+
+haldata = {"arch": ['FFFF'], 'name': ['Ucode']}
```

Comment:
the arch: needs to be transitioned to `[hal_base.HALBase.MfgIds.Any]` (or Intel)